[ONNX] Add type annotation for SARIF attributes

BowenBao opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #85107
* #85918
* __->__ #85898
* #85651
* #85428

Separated from #85651 to highlight the type annotation changes. It should support all type annotations
needed by SARIF, except for the dictionary types described verbally like the following example. For now it
is only annotated as `Any`. To enable it, we will need to extend `jschema_to_python` tool to allow passing
in type hints.
```json
        "messageStrings": {
          "description": "A set of name/value pairs with arbitrary names. Each value is a multiformatMessageString object, which holds message strings in plain text and (optionally) Markdown format. The strings can include placeholders, which can be used to construct a message in combination with an arbitrary number of additional string arguments.",
          "type": "object",
          "additionalProperties": {
            "$ref": "#/definitions/multiformatMessageString"
          }
        },
```

BowenBao committed:
[ONNX] Add type annotation for SARIF attributes

[ghstack-poisoned]

justinchuby reviewed (approved):
Is this done though an automated process?

BowenBao committed:
Update on "[ONNX] Add type annotation for SARIF attributes"


Separated from #85651 to highlight the type annotation changes. It should support all type annotations
needed by SARIF, except for the dictionary types described verbally like the following example. For now it
is only annotated as `Any`. To enable it, we will need to extend `jschema_to_python` tool to allow passing
in type hints.
```json
        "messageStrings": {
          "description": "A set of name/value pairs with arbitrary names. Each value is a multiformatMessageString object, which holds message strings in plain text and (optionally) Markdown format. The strings can include placeholders, which can be used to construct a message in combination with an arbitrary number of additional string arguments.",
          "type": "object",
          "additionalProperties": {
            "$ref": "#/definitions/multiformatMessageString"
          }
        },
```


[ghstack-poisoned]

BowenBao commented:
> Is this done though an automated process?

Yes, once we figure out how to properly distribute the alternated `jschema_to_python` generation tool, these are one click generated from `gen_sarif.sh`.

BowenBao committed:
Update on "[ONNX] Add type annotation for SARIF attributes"


Separated from #85651 to highlight the type annotation changes. It should support all type annotations
needed by SARIF, except for the dictionary types described verbally like the following example. For now it
is only annotated as `Any`. To enable it, we will need to extend `jschema_to_python` tool to allow passing
in type hints.
```json
        "messageStrings": {
          "description": "A set of name/value pairs with arbitrary names. Each value is a multiformatMessageString object, which holds message strings in plain text and (optionally) Markdown format. The strings can include placeholders, which can be used to construct a message in combination with an arbitrary number of additional string arguments.",
          "type": "object",
          "additionalProperties": {
            "$ref": "#/definitions/multiformatMessageString"
          }
        },
```


[ghstack-poisoned]

BowenBao commented:
@pytorchbot merge -g
